rfc/rfc-0002-bindings.md

```diff
@@ -0,0 +1,107 @@
+
+As a binding is backed by a ConfigMap and Secret a mechanism is needed to specify which ConfigMap and Secret. [ObjectReferences](https://godoc.org/k8s.io/api/core/v1#ObjectReference) are commonly used to define the coordinates to an arbitrary resources. A binding reference consists of metadata ObjectReference and a secret ObjectReference.
+
+Any resource that provides a binding should include the binding reference on its status, for example:
+
+```yaml
+name: provider
+...
+status:
+  ...
+  binding:
+    metadataRef:
+      apiVersion: v1
+      kind: ConfigMap
+      namespace: default
+      name: provider-binding-metadata
+    secretRef:
+      apiVersion: v1
+      kind: Secret
+      namespace: default
+      name: provider-binding-secret
+```
+
+The provider may change the values within the metadata and secret as needed, but should avoid changing the keys. The CNB Binding spec defines some well known keys that should be defined in the metadata.
+
+A workload consuming the binding can use the provider's binding, For example:
+
+```yaml
+name: consumer
+...
+spec:
+  ...
+  bindings:
+  - name: my-binding
+    metadataRef:
+      apiVersion: v1
+      kind: ConfigMap
+      namespace: default
+      name: provider-binding-metadata
+    secretRef:
+      apiVersion: v1
+      kind: Secret
+      namespace: default
+      name: provider-binding-secret
+```
+
+It is common to consume multiple bindings within a single workload. Each binding must have a unique name for the workload. The binding name is the directory the metadata and secret volumes are mounted within. While the name has no impact on the binding per se, workloads may use the name to convey semantic meaning about the intent of the binding (e.g. a primary database vs a replica).
+
+Note: The ConfigMap and Secret for the binding must be in the same namespace as the workload Pod, as Kubernetes does not support mounting resources defined in other namespaces.
+
+The reconciler for the resource consuming a binding is responsible for mapping the ConfigMap and Secret to volume mounts within the pod.
+
+kpack, as of v0.0.5, does not provide any mechanism for injecting volumes into builds, or any other means to satisfy the CNB Bindings spec. Further work will be required for proper integration of bindings at buildtime.
```

Comment:
kpack should provide a comparable consumer binding interface.

Reply:
We'd like to collaborate with the kpack team on how the Image resource consumes bindings.